### PR TITLE
Apply resolver depth and complexity limiter

### LINKED ISSUE #63: Add depth limiter to Apollo

### PR DESCRIPTION
**Pull request by norisalsaadie**
Closes #63 and #64

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -40,9 +42,24 @@ const init = async (): Promise<void> => {
...
+                    maximumComplexity: 1000,
+                    estimators: [
+                        // default fallback estimator.
+                        simpleEstimator({
+                            defaultComplexity: 1,
+                        }),
+                    ],
+                }),
+            ],
+            // @todo: Introspection queries will be blocked unless you are authenticated.
```
Comment (diversemix) [inline src/main.ts]: Can we make this switchable for Production? What do others think @will-byrne @erezmus @Cooryd

File: src/main.ts [hunk 2]
```diff
@@ -40,9 +42,24 @@ const init = async (): Promise<void> => {
...
+        const schema = makeExecutableSchema({ typeDefs, resolvers });
         const apolloServer = new ApolloServer({
-            typeDefs,
-            resolvers,
+            schema,
+            validationRules: [
+                depthLimit(5),
+                // may need to have resolver level complexity, but this is okay for now.
+                queryComplexity({
+                    maximumComplexity: 1000,
```
Comment (diversemix) [inline src/main.ts]: Looking here: https://typegraphql.ml/docs/0.17.0/complexity.html

I can see this has a `maximumComplexity: 8,  where as we have 1000 what does this number actually mean?

Comment (norisalsaadie) [reply src/main.ts]: We shouldn't make decision on this just yet as we don't have real queries yet. We should tweak it as we approach a realistic looking system. Perhaps the inline comment is not meaningful, I'll make it more explicit. I will reduce the default when switch to config.

File: src/main.ts (hunk 1 above)
Comment (norisalsaadie) [reply src/main.ts]: Sure

**Pull request merged**